### ISSUE TITLE
azurerm_linux_function_app_{,slot}: setting only site_config.cors.allowed_origins wants to add support_credentials = false

### ISSUE THREAD

**Issue by johanngyger** [hashicorp/terraform-provider-azurerm#20193]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Community Note

* Please vote on this issue by adding a :thumbsup: [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Terraform Version

1.3.7

### AzureRM Provider Version

3.40.0

### Affected Resource(s)/Data Source(s)

azurerm_linux_function_app, azurerm_linux_function_app_slot

### Terraform Configuration Files

```hcl
resource "azurerm_linux_function_app" "main" {
  (...)
  site_config {
    (...)
    cors {
      allowed_origins = var.cors_allowed_origins
    }
  }
}
```

### Debug Output/Panic Output

```shell
~ resource "azurerm_linux_function_app" "main" {
    ~ app_settings                       = (sensitive value)
      # (26 unchanged attributes hidden)
    ~ site_config {
          # (25 unchanged attributes hidden)
        + cors {
            + support_credentials = false
          }
          # (1 unchanged block hidden)
      }
      # (3 unchanged blocks hidden)
  }
```

### Expected Behaviour

No changes

### References

#20148

**Comment by xiaxyi** [comment]
Thanks for raising this issue @johanngyger . 

Can you help me to clarify about your TF config? Are you setting cors as below? 
```
 "cors": {
            "allowedOrigins": [],
            "supportCredentials": false
        },
```

**Comment by johanngyger** [comment]
Correct, @xiaxyi: it only happens with `allowedOrigins` being an empty set

**Comment by xiaxyi** [comment]
Yes, this is the default setting which equals to null value. We won't set the default setting in state so you are seeing diff here... I need to check whether to conditionally set the default value.

**Comment by johanngyger** [comment]
No trouble, the diff just keeps showing up. We have a workaround.

**Comment by xiaxyi** [comment]
I see. Are you choosing the ignore_change option to avoid the cors diff?

**Comment by johanngyger** [comment]
Yes, but I just realized I could use a dynamic block because we're using it in a module.

**Comment by SietseT** [comment]
This is also a problem with the `azurerm_windows_function_app` resource. Using `ignore_changes` doesn't work -- Terraform keeps seeing it as a change.

**Comment by matt-byrne** [comment]
Also seeing this issue. Even when I ignore `site_config[0].cors[0].support_credentials`, the change is re-planned every time.

**Comment by xiaxyi** [comment]
@matt-byrne ,Thanks for adding feedbacks, you can track the fix via https://github.com/hashicorp/terraform-provider-azurerm/pull/20251

**Comment by desolat** [comment]
Also with the supposed fix in azurerm 3.49.0 we are still seeing `support_credentials` showing up in the `plan` output, even when `site_config.cors` is not configured at all.

**Comment by matt-byrne** [comment]
Confirming I'm still seeing this issue in 3.49.0 also.
